Classes/Model/Dimension.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Networkteam\ImageProxy\Model;
+
+class Dimension
+{
+    protected int $width;
+    protected int $height;
+
+    public function __construct(?int $width, ?int $height)
+    {
+        $this->width = $width ?? 0;
+        $this->height = $height ?? 0;
+    }
+
+    public function getWidth(): int
+    {
+        return $this->width;
+    }
+
+    public function getHeight(): int
+    {
+        return $this->height;
+    }
+
+    public function getAspectRatio(): float
+    {
+        if ($this->height > 0) {
+            return $this->width / $this->height;
+        }
+
+        return 0;
+    }
+
+    public function isZero(): bool
+    {
+        return $this->noHeight() && $this->noWidth();
+    }
+
+    public function noWidth(): bool
+    {
+        return $this->width === 0;
+    }
+
+    public function noHeight(): bool
+    {
+        return $this->height === 0;
+    }
+
+    public function isGreater(Dimension $otherDimension) : bool
```

Comment:
I'm not sure this is always what we want: e.g. this dimension is very wide but not high, the other dimension might be not as wide but higher. Still `$this->isGreater($other)` might be `true`.

So maybe we need to check if either the width or the height is greater. But a test case showing this might be better ;)

Reply:
Fixed